deltacat/compute/compactor/compaction_session.py

```diff
@@ -56,12 +56,47 @@ def check_preconditions(
     )
 
 
+def check_build_initial_primary_key_indices_preconditions(
+            build_initial_primary_key_index: bool,
+            source_partition_locator: PartitionLocator,
+            source_compacted_partition_locator: PartitionLocator,
+            deltacat_storage):
+    if build_initial_primary_key_index:
+        assert source_compacted_partition_locator is not None
+    assert source_partition_locator.partition_values \
+        == source_compacted_partition_locator.partition_values, \
+        "Building initial primary key indices must use the same partition values " \
+        "for the source and compacted partition locator."
+    stream_locator = source_compacted_partition_locator.stream_locator
+    namespace = stream_locator.namespace
+    table_name = stream_locator.table_name
+    table_version = stream_locator.table_version
+    partition_values = source_compacted_partition_locator.partition_values
+    deltas_list_result = deltacat_storage.list_deltas(
+        namespace,
+        table_name,
+        partition_values,
+        table_version,
+        None,
+        None,
+        False,
+    )
+    delta_res = deltas_list_result.all_items()
+    assert len(delta_res) == 1, \
+        "Only one delta is expected when building initial primary key indices from compacted partition locator"
+    return delta_res[0].stream_position
+
+
 def compact_partition(
         source_partition_locator: PartitionLocator,
         compacted_partition_locator: PartitionLocator,
         primary_keys: Set[str],
         compaction_artifact_s3_bucket: str,
         last_stream_position_to_compact: int,
+        source_compacted_partition_locator: Optional[PartitionLocator] = None,
+        build_initial_primary_key_indices: Optional[bool] = False,
+        compaction_based_on_initial_primary_key_indices: Optional[bool] = False,
+        primary_key_source_stream_position: Optional[int] = None,
```

Comment:
Throughout this pkg, I often had to look into implementation to understand the purpose of a parameter. A documentation could ease readers in understanding at-least for this important method this is exposed.

Reply:
Thanks Raghavendra, will add explanation on these added arguments.